src/script/event/preprocessor/ReceiptsMiddleware.ts

```diff
@@ -49,10 +51,12 @@ export class ReceiptsMiddleware implements EventMiddleware {
       case ClientEvent.CONVERSATION.LOCATION:
       case ClientEvent.CONVERSATION.MESSAGE_ADD: {
         const qualifiedConversation = event.qualified_conversation || {domain: '', id: event.conversation};
-        return this.conversationRepository.getConversationById(qualifiedConversation).then(conversation => {
-          event.data.expects_read_confirmation = this.conversationRepository.expectReadReceipt(conversation);
-          return event;
-        });
+        const conversation = await this.conversationRepository.getConversationById(qualifiedConversation);
+        if (conversation?.isGroup()) {
```

Comment:
just point ting that out, any reason not to go back to
```suggestion
        if (conversation && conversation?.isGroup()) {
```

Reply:
nevermind was just a work around without the `conversation?` question mark